Project fails to build with Java 25

I can build the project with Java 21, but changing to JDK 25 results in errors.

```
$ java -version
openjdk version "25" 2025-09-16 LTS
OpenJDK Runtime Environment Temurin-25+36 (build 25+36-LTS)
OpenJDK 64-Bit Server VM Temurin-25+36 (build 25+36-LTS, mixed mode, sharing)
$ mvn --version
Apache Maven 3.9.11 (3e54c93a704957b63ee3494413a2b544fd3d825b)
Maven home: /Users/david/local/apache-maven
Java version: 25, vendor: Eclipse Adoptium, runtime: /Library/Java/JavaVirtualMachines/temurin-25.jdk/Contents/Home
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "15.7.1", arch: "aarch64", family: "mac"
```

Here's the error under Java 25:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.14.1:testCompile (default-testCompile) on project stork-core: Compilation failure: Compilation failure: 
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/AnchoredServiceDiscoveryProvider.java:[27,45] cannot find symbol
[ERROR]   symbol: class FakeConfiguration
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/AnchoredServiceDiscoveryProvider.java:[35,52] cannot find symbol
[ERROR]   symbol:   class FakeConfiguration
[ERROR]   location: class io.smallrye.stork.AnchoredServiceDiscoveryProvider
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/EmptyLoadBalancerProvider.java:[9,72] cannot find symbol
[ERROR]   symbol: class EmptySelectorConfiguration
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/EmptyLoadBalancerProvider.java:[12,44] cannot find symbol
[ERROR]   symbol:   class EmptySelectorConfiguration
[ERROR]   location: class io.smallrye.stork.EmptyLoadBalancerProvider
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/EmptyServiceDiscoveryProvider.java:[11,45] cannot find symbol
[ERROR]   symbol: class EmptyConfiguration
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/EmptyServiceDiscoveryProvider.java:[14,52] cannot find symbol
[ERROR]   symbol:   class EmptyConfiguration
[ERROR]   location: class io.smallrye.stork.EmptyServiceDiscoveryProvider
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/MockLoadBalancerProvider.java:[14,71] cannot find symbol
[ERROR]   symbol: class FakeSelectorConfiguration
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/MockLoadBalancerProvider.java:[17,44] cannot find symbol
[ERROR]   symbol:   class FakeSelectorConfiguration
[ERROR]   location: class io.smallrye.stork.MockLoadBalancerProvider
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/MockServiceDiscoveryProvider.java:[17,79] cannot find symbol
[ERROR]   symbol: class MockConfiguration
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/MockServiceDiscoveryProvider.java:[20,52] cannot find symbol
[ERROR]   symbol:   class MockConfiguration
[ERROR]   location: class io.smallrye.stork.MockServiceDiscoveryProvider
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/SingleLoadBalancerProvider.java:[10,73] cannot find symbol
[ERROR]   symbol: class SingleConfiguration
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/SingleLoadBalancerProvider.java:[12,44] cannot find symbol
[ERROR]   symbol:   class SingleConfiguration
[ERROR]   location: class io.smallrye.stork.SingleLoadBalancerProvider
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/TestServiceRegistrar.java:[13,19] cannot find symbol
[ERROR]   symbol:   class TestRegistrarConfiguration
[ERROR]   location: class io.smallrye.stork.TestServiceRegistrar
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/TestServiceRegistrar.java:[15,33] cannot find symbol
[ERROR]   symbol:   class TestRegistrarConfiguration
[ERROR]   location: class io.smallrye.stork.TestServiceRegistrar
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/TestServiceRegistrarProvider.java:[9,79] cannot find symbol
[ERROR]   symbol: class TestRegistrarConfiguration
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/TestServiceRegistrarProvider.java:[12,52] cannot find symbol
[ERROR]   symbol:   class TestRegistrarConfiguration
[ERROR]   location: class io.smallrye.stork.TestServiceRegistrarProvider
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/WeirdTypedLoadBalancerProvider.java:[12,41] cannot find symbol
[ERROR]   symbol: class TheseArentTheDroidsYoureLookingForSelectorConfiguration
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/WeirdTypedLoadBalancerProvider.java:[14,44] cannot find symbol
[ERROR]   symbol:   class TheseArentTheDroidsYoureLookingForSelectorConfiguration
[ERROR]   location: class io.smallrye.stork.WeirdTypedLoadBalancerProvider
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/WeirdTypedServiceDiscoveryProvider.java:[12,45] cannot find symbol
[ERROR]   symbol: class TheseArentTheDroidsYouAreLookingForConfiguration
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/WeirdTypedServiceDiscoveryProvider.java:[14,52] cannot find symbol
[ERROR]   symbol:   class TheseArentTheDroidsYouAreLookingForConfiguration
[ERROR]   location: class io.smallrye.stork.WeirdTypedServiceDiscoveryProvider
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/ObservationTest.java:[93,21] cannot find symbol
[ERROR]   symbol:   class MockConfiguration
[ERROR]   location: class io.smallrye.stork.ObservationTest
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/ObservationTest.java:[128,52] cannot find symbol
[ERROR]   symbol:   class FakeSelectorConfiguration
[ERROR]   location: class io.smallrye.stork.ObservationTest
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/ObservationTest.java:[219,21] cannot find symbol
[ERROR]   symbol:   class MockConfiguration
[ERROR]   location: class io.smallrye.stork.ObservationTest
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/ObservationTest.java:[252,52] cannot find symbol
[ERROR]   symbol:   class FakeSelectorConfiguration
[ERROR]   location: class io.smallrye.stork.ObservationTest
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/ProgrammaticApiTest.java:[44,42] cannot find symbol
[ERROR]   symbol:   class FakeConfiguration
[ERROR]   location: class io.smallrye.stork.ProgrammaticApiTest
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/ProgrammaticApiTest.java:[67,86] cannot find symbol
[ERROR]   symbol:   class SingleConfiguration
[ERROR]   location: class io.smallrye.stork.ProgrammaticApiTest
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/ProgrammaticApiTest.java:[67,42] cannot find symbol
[ERROR]   symbol:   class FakeConfiguration
[ERROR]   location: class io.smallrye.stork.ProgrammaticApiTest
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/ProgrammaticApiTest.java:[85,86] cannot find symbol
[ERROR]   symbol:   class SingleConfiguration
[ERROR]   location: class io.smallrye.stork.ProgrammaticApiTest
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/ProgrammaticApiTest.java:[85,42] cannot find symbol
[ERROR]   symbol:   class FakeConfiguration
[ERROR]   location: class io.smallrye.stork.ProgrammaticApiTest
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/ProgrammaticApiTest.java:[89,48] cannot find symbol
[ERROR]   symbol:   class SingleConfiguration
[ERROR]   location: class io.smallrye.stork.ProgrammaticApiTest
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/StorkWithCDITest.java:[341,27] cannot find symbol
[ERROR]   symbol:   class AnchoredServiceDiscoveryProviderLoader
[ERROR]   location: class io.smallrye.stork.StorkWithCDITest
[ERROR] /Volumes/Case-sensitive/david/src/java/smallrye-stork/core/src/test/java/io/smallrye/stork/StorkWithCDITest.java:[342,27] cannot find symbol
[ERROR]   symbol:   class MockLoadBalancerProviderLoader
[ERROR]   location: class io.smallrye.stork.StorkWithCDITest
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :stork-core
```

--- a/core/pom.xml
+++ b/core/pom.xml
@@ -90,6 +90,17 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+<!--                    This makes the processor available for both compile and test-compile phases, needed here for fixing missing generated classes when using jdk 25-->
+                    <annotationProcessorPathsUseDepMgmt>true</annotationProcessorPathsUseDepMgmt>
+                    <annotationProcessorPaths>
+                        <path>
+                            <groupId>io.smallrye.stork</groupId>
+                            <artifactId>stork-configuration-generator</artifactId>
+                            <version>${project.version}</version>
+                        </path>
+                    </annotationProcessorPaths>
+                </configuration>
                 <executions>
                     <execution>
                         <id>default-compile</id>

--- a/pom.xml
+++ b/pom.xml
@@ -304,30 +304,6 @@
                         <excludePackageNames>*.impl:*.impl.*</excludePackageNames>
                     </configuration>
                 </plugin>
-                <plugin>
-                    <groupId>org.apache.maven.plugins</groupId>
-                    <artifactId>maven-compiler-plugin</artifactId>
-                    <version>${plugin.version.compiler}</version>
-                    <executions>
-                        <execution>
-                            <id>default-compile</id>
-                            <phase>compile</phase>
-                            <goals>
-                                <goal>compile</goal>
-                            </goals>
-                            <configuration>
-                                <annotationProcessorPathsUseDepMgmt>true</annotationProcessorPathsUseDepMgmt>
-                                <annotationProcessorPaths>
-                                    <path>
-                                        <groupId>io.smallrye.stork</groupId>
-                                        <artifactId>stork-configuration-generator</artifactId>
-                                        <version>${project.version}</version>
-                                    </path>
-                                </annotationProcessorPaths>
-                            </configuration>
-                        </execution>
-                    </executions>
-                </plugin>
             </plugins>
         </pluginManagement>
         <plugins>
@@ -389,7 +365,9 @@
         </plugins>
     </build>
 
+
     <modules>
+        <module>config-generator</module>
         <module>api</module>
         <module>bom</module>
         <module>core</module>
@@ -412,7 +390,6 @@
         <module>service-registration/eureka</module>
         <module>test-utils</module>
         <module>docs</module>
-        <module>config-generator</module>
     </modules>
 
     <profiles>

